### PR TITLE
Recursive Methods: Add link to corrected solution

### LINKED ISSUE #27154: Recursive Methods: Solution in assignment is wrong
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

The solution provided for question 6 in the [article](https://www.codingame.com/playgrounds/5422/js-interview-prep-recursion) linked in the "Test Yourself" section seems wrong to me. It fails the following test case:-
```js
var nestedObject = {
  data: {
    a: { value: 1 },
    b: { value: 2 },
  },
};

console.log(contains(nestedObject, 2)); // should be true, returns false

function contains(obj, value){
	for(var key in obj){
		if(typeof obj[key] === 'object'){
			return contains(obj[key], value);
		}

		if (obj[key] === value){
			return true;
		}
	}
	return false;
}
```

Correct code:
```js
function contains(object, value) {
  if (typeof object != "object") {
    return object === value;
  }
  for (const key in object) {
    if (contains(object[key], value)) {
      return true;
    }
  }
  return false;
}
```

Maybe a note should be added with the correct code?

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/javascript-recursive-methods

### (Optional) Discord Name

bonothebonobo

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by BinoManjesh**
## Because

Question 6 of the article linked in the "Test yourself" section was incorrect

## This PR

- Add note to explain that solution is incorrect
- Add link to corrected solution

## Issue

Closes #27154

## Pull Request Requirements
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: javascript/computer_science/recursive_methods.md [hunk 1]
```diff
@@ -34,6 +34,11 @@ This section contains a general overview of topics that you will learn in this l
 
 <div class="lesson-content__panel" markdown="1">
   1. Solve each of the questions from this [Code Quiz](https://www.codingame.com/playgrounds/5422/js-interview-prep-recursion) on Recursion. It is not important to have recursive algorithms committed to memory at this point; just understand how to create and use them.
+
+<div class="lesson-note lesson-note--warning" markdown="1">
+The solution for "Question 6: Search JS object" is incorrect. See the corrected solution [here](https://gist.github.com/JoshDevHub/b00125f483d4a1ecc257eaa030916973) after you solve it.
```
Comment (JoshDevHub) [inline javascript/computer_science/recursive_methods.md:39]:
```suggestion
The solution for "Question 6: Search JS object" is incorrect. See the [corrected solution](https://gist.github.com/JoshDevHub/b00125f483d4a1ecc257eaa030916973) after you solve it.
```
Going to recommend this small change. This better keeps to our [style guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#links) regarding links.

**Review by JoshDevHub — CHANGES_REQUESTED** [review]
Thank you for doing this, and nice job finding the problem in the given solution.

Going to recommend one small change with the wording of the link.

Also, this file you're working in has a few errors in it that aren't passing our Markdownlint tooling. These weren't introduced by you, but would you be willing to fix them? They're all related to extra spaces appearing after bullet points in a list.
```
- single space after marker is good

-  double space after marker is bad
```
If you don't want to, I can just merge and leave it for someone else. But thought I'd let you take it on if you want.

**Comment by BinoManjesh** [comment]
Ok i'll fix the markdown errors

**Comment by BinoManjesh** [comment]
@JoshDevHub I've added your suggestion and fixed the markdown errors

**Review by JoshDevHub — APPROVED** [review]
Awesome work. Thank you for fixing the lint errors :raised_hands:

**Pull request merged**